packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,436 @@
+///         expired.
+///         If a challenge is expired, the challenger's bond is unlocked and the challenged commitment is added to the
+///         chain of expired challenges.
+contract DataAvailabilityChallenge is OwnableUpgradeable, ISemver {
+    /// @notice Error for when the provided resolver refund percentage exceeds 100%.
+    error InvalidResolverRefundPercentage(uint256 invalidResolverRefundPercentage);
+
+    /// @notice Error for when the challenger's bond is too low.
+    error BondTooLow(uint256 balance, uint256 required);
+
+    /// @notice Error for when attempting to challenge a commitment that already has a challenge.
+    error ChallengeExists();
+
+    /// @notice Error for when attempting to resolve a challenge that is not active.
+    error ChallengeNotActive();
+
+    /// @notice Error for when attempting to unlock a bond from a challenge that is not expired.
+    error ChallengeNotExpired();
+
+    /// @notice Error for when attempting to challenge a commitment that is not in the challenge window.
+    error ChallengeWindowNotOpen();
+
+    /// @notice Error for when the provided input data doesn't match the commitment.
+    error InvalidInputData(bytes providedDataCommitment, bytes expectedCommitment);
+
+    /// @notice Error for when the call to withdraw a bond failed.
+    error WithdrawalFailed();
+
+    /// @notice Error for when a the type of a given commitment is unknown
+    error UnknownCommitmentType(bytes1 commitmentType);
+
+    /// @notice Error for when the commitment length does not match the commitment type
+    error InvalidCommitmentLength(uint8 commitmentType, uint256 expectedLength, uint256 actualLength);
+
+    /// @notice An event that is emitted when the status of a challenge changes.
+    /// @param challengedCommitment The commitment that is being challenged.
+    /// @param challengedBlockNumber The block number at which the commitment was made.
+    /// @param status The new status of the challenge.
+    event ChallengeStatusChanged(
+        uint256 indexed challengedBlockNumber, bytes challengedCommitment, ChallengeStatus status
+    );
+
+    /// @notice An event that is emitted when the bond size required to initiate a challenge changes.
+    event RequiredBondSizeChanged(uint256 challengeWindow);
+
+    /// @notice An event that is emitted when the percentage of the resolving cost to be refunded to the resolver
+    /// changes.
+    event ResolverRefundPercentageChanged(uint256 resolverRefundPercentage);
+
+    /// @notice An event that is emitted when a user's bond balance changes.
+    event BalanceChanged(address account, uint256 balance);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice The fixed cost of resolving a challenge.
+    /// @dev The value is estimated by measuring the cost of resolving with `bytes(0)`
```

Comment:
Shouldn't this be based on worst case and not best case?

Reply:
The fixed cost is the cost that is independent of the calldata size, the variable cost of ~16,64 gas per byte of calldata in the resolution is the worst case cost assuming every byte of calldata is non-zero (see variable below)